spec/unit/util/run_mode_spec.rb

```diff
@@ -2,6 +2,13 @@
 require 'spec_helper'
 
 describe Puppet::Util::RunMode do
+
+  # Discriminator for tests that attempts to unset HOME since that, for reasons currently unknown,
```

Comment:
@Iristyle, Great job finding the actual issue. I tried but didn't succeed. I agree that it's a good thing to understand the problem. My intention was not to paper over it, but rather that the current tests could be made into acceptance tests instead. I suggest that in the commit comment.